be/src/formats/parquet/group_reader.cpp

```diff
@@ -170,6 +172,131 @@ Status GroupReader::get_next(ChunkPtr* chunk, size_t* row_count) {
     return status;
 }
 
+Status GroupReader::_do_get_next_new(ChunkPtr* chunk, size_t* row_count) {
+    if (_is_group_filtered) {
+        *row_count = 0;
+        return Status::EndOfFile("");
+    }
+
+    _read_chunk->reset();
+
+    ChunkPtr active_chunk = _create_read_chunk(_active_column_indices);
+    // to complicity with _do_get_next will break and return even active_row is all filtered.
+    // but a better choice is don't return until really have some results.
+    while (true) {
+        if (!_range_iter.has_more()) {
+            *row_count = 0;
+            return Status::EndOfFile("");
+        }
+
+        auto r = _range_iter.next(*row_count);
+        auto count = r.span_size();
+        _param.stats->raw_rows_read += count;
+
+        bool has_filter = false;
+        Filter chunk_filter(count, 1);
+
+        // row id filter
+        if ((nullptr != _need_skip_rowids) && !_need_skip_rowids->empty()) {
+            {
+                SCOPED_RAW_TIMER(&_param.stats->build_iceberg_pos_filter_ns);
+                auto start_str = _need_skip_rowids->lower_bound(r.begin());
+                auto end_str = _need_skip_rowids->upper_bound(r.end() - 1);
+
+                for (; start_str != end_str; start_str++) {
+                    chunk_filter[*start_str - r.begin()] = 0;
+                    has_filter = true;
+                }
+            }
+        }
+
+        if (has_filter) {
+            RETURN_IF_ERROR(_read_range(_active_column_indices, r, &chunk_filter, &active_chunk));
+        } else {
+            RETURN_IF_ERROR(_read_range(_active_column_indices, r, nullptr, &active_chunk));
+        }
+
+        // dict filter chunk
```

Comment:
why don't we filter per active column round by round

Reply:
Will do it later.

Reply:
will in another pr